src/main/java/uk/gov/ch/dao/ShareholderDao.java

```diff
@@ -0,0 +1,77 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OracleQueryApplication.APPLICATION_NAME_SPACE);
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    static final String SHAREHOLDER_COUNT_SQL = "SELECT COUNT(*) FROM shareholder sh INNER JOIN shareholding shd ON sh.shareholder_id = shd.shareholding_id JOIN corporate_body cb on cb.corporate_body_id = shd.corporate_body_id WHERE cb.incorporation_number = ?";
+    static final String SHAREHOLDER_ELECTED_COUNT_SQL = "SELECT COUNT(*) FROM shareholder_elected sh JOIN sholdingelc_sholderelc_link shl ON sh.shareholder_elected_id = shl.shareholder_elected_id  INNER JOIN shareholding_elected shd ON shd.shareholding_elected_id = shl.shareholding_elected_id JOIN corporate_body cb on cb.corporate_body_id = shd.corporate_body_id WHERE cb.incorporation_number = ?";
+    
+    static final String SHAREHOLDER_LIST_SQL = "SELECT sh.shareholder_forename_1 as forename1, sh.shareholder_forename_2 as forename2, sh.shareholder_surname as surname, "
+        + "sh.address_id as addressId, shd.number_of_shares as shares, shd.share_class_type_id as shareClassTypeId, shd.currency_type_id as currencyTypeId "
+        + "FROM shareholder sh INNER JOIN shareholding shd ON sh.shareholder_id = shd.shareholding_id "
+        + "JOIN corporate_body cb on cb.corporate_body_id = shd.corporate_body_id WHERE cb.incorporation_number = ?";
+
+    static final String SHAREHOLDER_ELECTED_LIST_SQL = "SELECT sh.shareholder_elected_forename_1 as forename1, sh.shareholder_elected_forename_2 as forename2, sh.shareholder_elected_surname as surname, "
+        + "sh.address_id as addressId, shd.number_of_shares as shares, shd.share_class_type_id as shareClassTypeId, shd.currency_type_id as currencyTypeId "
+        + "FROM shareholder_elected sh JOIN sholdingelc_sholderelc_link shl ON sh.shareholder_elected_id = shl.shareholder_elected_id "
+        + "INNER JOIN shareholding_elected shd ON shd.shareholding_elected_id = shl.shareholding_elected_id "
+        + "JOIN corporate_body cb on cb.corporate_body_id = shd.corporate_body_id WHERE cb.incorporation_number = ?";
+
+    public List<Shareholder> getShareholders(String incorporationNumber) {
+        List<Shareholder> list = jdbcTemplate.query(SHAREHOLDER_LIST_SQL, getParam(incorporationNumber), new BeanPropertyRowMapper<>(Shareholder.class));
+
+        LOGGER.info("Returned " + list.size() + " shareholders from SHAREHOLDER tables.");
+
+        if (list.isEmpty()) {
+            list = jdbcTemplate.query(SHAREHOLDER_ELECTED_LIST_SQL, getParam(incorporationNumber), new BeanPropertyRowMapper<>(Shareholder.class));
+
+            LOGGER.info("Returned " + list.size() + " shareholders from SHAREHOLDER_ELECTED tables.");
+        }
+
+        return list;
+    }
+
+    public int getShareholderCount(String incorporationNumber) {
+        int count;
+        count = jdbcTemplate.queryForObject(SHAREHOLDER_COUNT_SQL, Integer.class, incorporationNumber);
```

Comment:
count can be declared and assigned on one line here eg
int count = jdbcTemplate.queryForObject(.....